src/directory/managed_directory.rs

```diff
@@ -222,9 +223,21 @@ impl ManagedDirectory {
             .write()
             .expect("Managed file lock poisoned");
         let has_changed = meta_wlock.managed_paths.insert(filepath.to_owned());
-        if has_changed {
-            save_managed_paths(self.directory.as_ref(), &meta_wlock)?;
+        if !has_changed {
+            return Ok(());
+        }
+        save_managed_paths(self.directory.as_ref(), &meta_wlock)?;
+        if meta_wlock.managed_paths.len() > 1 {
```

Comment:
being explicit here would increase readability. It would also remove the dependency to `meta_wlock.managed_paths`. Could have unexpected side effects when someone alters `meta_wlock.managed_paths` elsewhere.

if meta_json_already_synced_to_directory {

Reply:
Good point

Reply:
I'll keep the lock though. It makes it easy to reason with. Of course it might poison it, but here if we fail a write, I think halting and catch fire is a decent policy.